docs/content/blog/authelia-traefik-setup-guide/index.md

```diff
@@ -0,0 +1,391 @@
+date: 2024-11-23T10:10:09+10:00
+draft: false
+weight: 50
+categories: ["Guides"]
+tags: ["guides"]
+contributors: ["Brynn Crowley"]
+pinned: false
+homepage: false
+seo:
+  title: "" # custom title (optional)
+  description: "" # custom description (recommended)
+  canonical: "" # custom canonical URL (optional)
+  noindex: false # false (default) or true
+---
+This is not a demo. If you would like an all-in-one demo, please take a look at our [local bundle](https://www.authelia.com/integration/deployment/docker/#local).
+## Assumptions and Adaptation
+This guide makes a few assumptions. These assumptions may require adaptation in more advanced and complex scenarios. We can not reasonably have examples for every advanced configuration option that exists. Some of these values can be automatically replaced with documentation variables.
+
+{{< sitevar-preferences >}}
+
+We make the following assumptions:
+- [Docker](https://docs.docker.com/engine/install/) is configured correctly.
+- Single Host
+- You will have to adapt all instances of `{{< sitevar name="host" nojs="authelia" >}}` in the URL if:
+  - you're using a different container name
+  - you deployed the proxy to a different location
+- You will have to adapt all instances of `{{< sitevar name="port" nojs="9091" >}}` in the URL if:
+  - you have adjusted the default port in the configuration
+- You will have to adapt the entire URL if:
+  - Authelia is on a different host to the proxy
+- All services are part of the `{{< sitevar name="domain" nojs="example.com" >}}` domain:
+- This domain and the subdomains will have to be adapted in all examples to match your specific domains unless you're
+  just testing, or you want to use that specific domain
+
+## File Structure
+The first thing we want to do is set up the file structure. Which should look something like this:
+```text
+📁 project
+ ┣ 📁 authelia
+ ┃  ┣ 📁 config
+ ┃  ┃ ┣ 📄 configuration.yml
+ ┃  ┃ ┗ 📄 users.yml
+ ┃  ┗ 📁 secrets
+ ┣ 📄 compose.yml
+ ┗ 📁 traefik
+    ┣ 📁 config
+    ┃ ┣ 📄 dynamic.yml
+    ┃ ┗ 📄 traefik.yml
+    ┣ 📁 data
+    ┃ ┗ 📄 acme.json
+    ┣ 📁 logs
+    ┗ 📁 secrets
+```
+
+## Traefik and Whoami
+{{< callout context="note" title="Note" icon="outline/info-circle" >}}
+We'll focus on the minimal configuration needed to work with Authelia. For advanced Traefik features and configurations, consult their documentation.
+{{< /callout >}}
+
+Next, we'll set up Traefik as our reverse proxy. For detailed Traefik documentation, refer to the [official Traefik docs](https://doc.traefik.io/traefik/).
+
+#### Docker Compose
+
+
+```yaml{title="compose.yml"}
+services:
+  traefik:
+    image: traefik:latest
+    container_name: traefik
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges=true
+    networks:
+      proxy: {}
+      authelia:
+        aliases:
+          - '{{< sitevar name="subdomain-authelia" nojs="auth" >}}.{{< sitevar name="domain" nojs="example.com" >}}'
```

Comment:
```suggestion
      proxy:
        aliases:
          - '{{< sitevar name="subdomain-authelia" nojs="auth" >}}.{{< sitevar name="domain" nojs="example.com" >}}'
      authelia: {}
```

Reply:
It should be noted that the Authelia container is not a member of the `proxy` network thus the alias on the `proxy` network would be ineffectual.

The alias should however be on the Authelia container not on the Traefik container.

Reply:
It's actually the other way around. The alias exists for other containers on the same network as the proxy.